src/main/java/org/spongepowered/api/map/MapCanvas.java

```diff
@@ -0,0 +1,128 @@
+import org.spongepowered.api.map.font.MapFont;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.util.Color;
+
+import java.awt.Image;
+
+/**
+ * A representation of the canvas (pixel surface) of a map.
+ */
+public interface MapCanvas {
+
+    /**
+     * Returns the width of the map's canvas.
+     *
+     * @return The map canvas's width
+     */
+    int getWidth();
+
+    /**
+     * Returns the height of the map's canvas.
+     *
+     * @return The map canvas's height
+     */
+    int getHeight();
+
+    /**
+     * Sets the specified pixel to the exact {@link MapColor} given.
+     *
+     * <p>The coordinates start with 0,0 as the top left corner moving right
+     * and down.</p>
+     *
+     * @param x The x coordinate of the pixel
+     * @param y The y coordinate of the pixel
+     * @param color The color to set the pixel to
+     * @throws IndexOutOfBoundsException If the coordinates are greater than
+     *         the width and height of the map's canvas
+     */
+    void setPixelExact(int x, int y, MapColor color);
+
+    /**
+     * Sets the specified pixel to the {@link MapColor} that is most similar to the {@link Color} provided.
+     *
+     * <p>The coordinates start with 0,0 as the top left corner moving right
+     * and down.</p>
+     *
+     * @param x The x coordinate of the pixel
+     * @param y The y coordinate of the pixel
+     * @param color The color to be matched before setting the pixel to the closest {@link MapColor}
+     * @throws IndexOutOfBoundsException If the coordinates are greater than
+     *         the width and height of the map's canvas
+     */
+    void setPixelSimilar(int x, int y, Color color);
+
+    /**
+     * Returns the {@link MapColor} at the specified pixel. An unset pixel will be returned as
+     * {@link org.spongepowered.api.map.color.MapPalette#AIR}
+     *
+     * The coordinates start with 0,0 as the top left corner moving right and
+     * down.
+     *
+     * @param x The x coordinate of the pixel to get
+     * @param y The y coordinate of the pixel to get
+     * @return The color of the pixel at the specified coordinates
+     * @throws IndexOutOfBoundsException If the coordinates are greater than or equal to
+     * the width and height of the map's canvas
+     */
+    MapColor getPixel(int x, int y);
+
+    /**
+     * Create an {@link Image} out of the map allowing it to be viewed outside
+     * the game in conventional image viewers and shared on the web. Simply
+     * converts indexes in the map's canvas to the RGB values they represent.
+     *
+     * @return The rendered image
+     */
+    Image toImage();
+
+    /**
+     * Draws an image to the map, clipping it if it goes out of bounds.
+     *
+     * @param x The top left x coordinate on the map
+     * @param y The top left y coordinate on the map
+     * @param image The image to draw
+     */
+    void drawImage(int x, int y, Image image);
```

Comment:
Maybe add a defaulted method that inserts the Image at 0,0 since this will usually ve the usecase.
`drawImage(Image)`